### ISSUE TITLE
[Privacy]: User added labels/strings are sent to segment.io

### ISSUE THREAD

**Issue by MicahZoltu** [MetaMask/metamask-extension#16754]
### Describe the bug

When a user adds a custom RPC endpoint, they are given the opportunity to label it with a custom string.  This string is sent to segment.io which allows for incredibly strong fingerprinting of a user and correlation between their different browsers/profiles/devices as an individual user will likely use similar labeling scheme across devices.

### Steps to reproduce

1. Add a custom network.
2. Wait for analytics to be sent to segment.io.
3. Notice that it sends the label you gave for the custom network to segment.io.

### Error messages or log output

```shell
I believe this is the line that needs to be removed.
https://github.com/MetaMask/metamask-extension/blob/80e4a1fef4b8e7561bf222e4f5ab4e9de3e235cc/app/scripts/metamask-controller.js#L2088
```

### Version

All versions since segment.io was added.

### Browser

Brave

### Operating system

MacOS

### Additional context

This was discussed a bit in https://github.com/MetaMask/metamask-extension/pull/16710#discussion_r1034237131 but that is closed to external contributors so it isn't possible to comment there.  Here is the last comment in that PR on the issue:

> network name is not leaking any sensitive data, and in addition can be useful to know based on whether network was named by the user (via custom form) or by the addEthereumChain API. For now i'll remove block_explorer_url. i was going to do it in a different PR but this one is fine.

While it is not leaking secrets like previously, it *is* a privacy leak because it allows correlation of a single user across multiple installs of MetaMask across multiple devices as a user is likely to use the same naming convention in multiple places.

It is incredibly common that information is useful for analytics, but user privacy should trump that, even if it makes it harder for the developers to troubleshoot/debug/improve.  Generally speaking, anything where the user is adding custom strings, images, etc. is a very strong fingerprinting opportunity and should be avoided whenever possible.

**Comment by brad-decker** [comment]
@MicahZoltu we discussed this issue outside of github after that conversation on the PR. The only reason knowing the name would have been useful for us would be in situations where an unscrupulous actor attempted to confuse users by suggesting networks with mismatched names/chain ids/rpc urls such that the user is confused into thinking they are operating on a trusted network and are in fact operating on a nefarious one. However, the ability to see this would only be effective if we are tracking rpc_url which we removed for the reasons you outlined in #15170 . 

TL:DR; we agree and we are going to remove it in a forthcoming pull request. cc @legobeat @worldlyjohn 

My apologies for my comment on the pull request being dismissive of the idea. My intention was to get the arguably higher priority item removed, not to shut down conversation on further improvements that could be made.
